boards: arm: nrf52840_pca10059: Add pwmleds to DT

nullr0ute opened the pull request:
Add a pwm0 entry and PWM support for the RGB LED so that the
rgb_led and fade_led demos work on the nrf52840_pca10059.

Signed-off-by: Peter Robinson <pbrobinson@gmail.com>

anangl commented on boards/arm/nrf52840_pca10059/nrf52840_pca10059.dts:
```diff
@@ -107,6 +126,15 @@
 
+&pwm0 {
+	status = "okay";
+	ch0-pin = <8>;
```
This channel drives `led_1`, which is [active low](https://github.com/zephyrproject-rtos/zephyr/pull/22695/files#diff-6a7a06ff03ddb2a64d4a13dd7186ed9eL34), so you need to add `ch0-inverted` here.

nullr0ute replied:
I did actually try that as that's what I thought it should be but then you never got any red as it cycled through.

anangl replied:
> I did actually try that as that's what I thought it should be but then you never got any red as it cycled through.

@nullr0ute Well, you will see red in the rgb_led sample when the channel 0 is inverted. You just need to wait 100 seconds (then the red LED is driven with a non-zero width pulse for the first time) for it... Or you can shorten the [sleeping period](https://github.com/zephyrproject-rtos/zephyr/blob/master/samples/basic/rgb_led/src/main.c#L107), to see red earlier. It seems that the sample could be improved, to better present RGB LEDs in action.
When the channel is not inverted, you see red from the beginning, because then the red LED starts from being on and is (very) slowly faded off.

anangl replied:
@nullr0ute See also https://github.com/zephyrproject-rtos/zephyr/pull/22719. You can try with the fade_led sample after applying this patch to see that the channel for the red LED should be inverted.

nullr0ute committed:
boards: arm: nrf52840_pca10059: Add pwmleds to DT

Add a pwm0 entry and PWM support for the RGB LED so that the
rgb_led and fade_led demos work on the nrf52840_pca10059.

Signed-off-by: Peter Robinson <pbrobinson@gmail.com>

lemrey reviewed (approved):
LGTM

ioannisg commented:
@nullr0ute could you rebase this PR, so it re-runs CI ?

jhedberg commented:
> @nullr0ute could you rebase this PR, so it re-runs CI ?

@ioannisg I triggered a re-run from within Shippable

jhedberg commented:
If I interpret the Shippable failure logs correctly, it seems that #23101 (which was merged today) is to blame?

jhedberg commented:
> If I interpret the Shippable failure logs correctly, it seems that #23101 (which was merged today) is to blame?

Created a hotfix: #23377
